README.rst

```diff
@@ -233,14 +233,11 @@ Releasing
 
 In order to deploy to production:
 
-1. Go to `deployment workflow page <https://github.com/mozilla-sre-deploy/deploy-remote-settings/actions/workflows/gcp.yaml>`_
```

Comment:
Should we mention that deployments to prod are triggered automatically upon tag creation and this is only needed for a manual workflow? I'm not sure how to word/organize this in the most effective way.

Reply:
> Should we mention that deployments to prod are triggered automatically upon tag creation

That isn't the case anymore -- for now, anyway. See [this PR's description](https://github.com/mozilla-sre-deploy/deploy-remote-settings/pull/19)

Reply:
Oh thank you, I missed that prod was fully manual now.